x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -371,9 +378,34 @@ func (k Keeper) ComputeNextValidators(ctx sdk.Context, consumerId string, bonded
 
 	nextValidators := k.FilterValidators(ctx, consumerId, bondedValidators,
 		func(providerAddr types.ProviderConsAddress) bool {
-			return k.CanValidateChain(ctx, consumerId, providerAddr) && k.FulfillsMinStake(ctx, consumerId, providerAddr)
+			return k.CanValidateChain(ctx, consumerId, providerAddr, minPowerToOptIn) && k.FulfillsMinStake(ctx, consumerId, providerAddr)
 		})
 
 	nextValidators = k.CapValidatorSet(ctx, consumerId, nextValidators)
 	return k.CapValidatorsPower(ctx, consumerId, nextValidators)
 }
+
+// HasMinPower returns true if the `providerAddr` voting power is GTE than the given minimum power
+func (k Keeper) HasMinPower(ctx sdk.Context, providerAddr types.ProviderConsAddress, minPower uint64) bool {
+	val, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.Address)
+	if err != nil {
+		k.Logger(ctx).Error("cannot get last validator power %s: %s", providerAddr, err)
```

Comment:
Can we use format specifiers (e.g., `%s`) inside `.Error`?
Here and below.

Reply:
Updated thanks.